### PR TITLE
Login page and auth

### PR DESCRIPTION
**Pull request by andykwan1124**
For reference: https://github.com/nwplus/CMS-V2/pull/22

### CODE REVIEW

File: pages/index.js [hunk 1]
```diff
@@ -1,27 +1,109 @@
...
+    width: 100vw;
+`
+
+const LeftContainer = styled.div`
+    display: flex;
+    align-items: center;
+    flex-direction: column;
+    height: 100vh;
+    width: 337px;
+    background-color: white;
```
Comment (panjenny0) [inline pages/index.js:24]:
```suggestion
    background-color: ${COLOR.WHITE};
```

File: pages/index.js [hunk 2]
```diff
@@ -1,27 +1,109 @@
...
+      if (isAdmin) {
+        await router.push('/landing')
+      } else {
+        const res = await setAdmin()
+        if (res.data.isAdmin) {
+          await user.getIdToken(true)
+          await router.push('/landing')
+        } else {
+          await firebase.auth().signOut()
+          await router.push('/')
```
Comment (panjenny0) [inline pages/index.js:78]: I think `await` before `router.push` is not needed
```suggestion
        router.push('/landing')
      } else {
        const res = await setAdmin()
        if (res.data.isAdmin) {
          await user.getIdToken(true)
          router.push('/landing')
        } else {
          await firebase.auth().signOut()
          router.push('/')
```

File: utility/auth.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        const unsubscribe = firebase.auth().onAuthStateChanged(async user => {
+            if (user === null) {
+                await router.replace('/')
+            } else {
+                const isAdmin = await checkAdminClaim(user)
+                if (!isAdmin) {
+                    await router.replace('/')
+                } else {
+                    setUser(user)
+                    if (router.pathname=='/') await router.push('/landing')
```
Comment (panjenny0) [inline utility/auth.js:39]:
```suggestion
                    router.replace('/')
                } else {
                    setUser(user)
                    if (router.pathname === '/') router.push('/landing')
```

**Review by panjenny0 — CHANGES_REQUESTED** [review]

File: pages/index.js (hunk 2 above)
Comment (andykwan1124) [reply pages/index.js]: You're right that I don't need the `await` in the login page, but I think I should still use it in auth.js. It seems like `router.push` is an asynchronous operation, so if a user that is not logged in tries to access the landing page, `setIsLoading(false)` might run first before `router.push` finishes pushing that user back to the login page, which results in a brief flash of protected content.

Comment (panjenny0) [reply pages/index.js]: Oh ok makes sense, gotcha 👍

**Review by panjenny0 — APPROVED** [review]
lgtm!

**Review by Ciuffi — APPROVED** [review]
Works great, good job andy!

**Pull request merged**